plugins/linux/ifplugin/interface_config.go

```diff
@@ -843,6 +847,80 @@ func (c *LinuxInterfaceConfigurator) removeFromCache(iface *interfaces.LinuxInte
 	return nil
 }
 
+// Watcher receives events from state updater about created/removed linux interfaces and performs appropriate actions
+func (c *LinuxInterfaceConfigurator) watchLinuxStateUpdater() {
+	c.log.Debugf("Linux interface state watcher started")
+
+	for {
+		linuxIf := <-c.ifNotif
+		// Ignore empty events
+		if linuxIf == nil {
```

Comment:
This currently cannot happen if I'm not mistaken. The only way I can think of is if the channel is closed (which isn't being done and perhaps should) and it that case you should definitely check it's closed and return from this function, otherwise on close this will endlessly continue the loop, since closed channel returns immediately.